diff -up akonadi-19.11.80/src/server/storage/itemretrievalmanager.h.omv~ akonadi-19.11.80/src/server/storage/itemretrievalmanager.h
--- akonadi-19.11.80/src/server/storage/itemretrievalmanager.h.omv~	2019-11-15 20:53:04.533697770 +0100
+++ akonadi-19.11.80/src/server/storage/itemretrievalmanager.h	2019-11-15 20:53:07.017705216 +0100
@@ -22,7 +22,6 @@
 
 #include "itemretriever.h"
 #include "akthread.h"
-#include <shared/akstd.h>
 
 #include <QHash>
 #include <QObject>
diff -up akonadi-19.11.80/src/shared/akstd.h.omv~ akonadi-19.11.80/src/shared/akstd.h
